orion/handlers/http/defaults.go

```diff
@@ -41,14 +42,20 @@ func DefaultEncoder(req *http.Request, r interface{}) error {
 }
 
 func deserialize(ctx context.Context, data []byte, r interface{}) error {
-	serType := ContentTypeFromHeaders(ctx)
+	serType := strings.TrimSpace(ContentTypeFromHeaders(ctx))
+	if strings.Index(serType, ";") > -1 {
+		// patch 1: to prevent cases like "application/json; charset=utf-8"
+		serType = strings.TrimSpace(strings.Split(serType, ";")[0])
+	}
+
 	switch serType {
 	case modifiers.ProtoBuf:
 		if protoReq, ok := r.(proto.Message); ok {
 			return proto.Unmarshal(data, protoReq)
 		}
 		fallthrough
-	case modifiers.JSONPB:
+	case modifiers.JSONPB, modifiers.JSON, "":
```

Comment:
previously modifiers.JSON and "" will go to default section, what's the intent to make jsonpb as default?

Reply:
We have seen issue in client handling serialization with Enum, basically it has to treat enum as number instead of string like jsonpb does, which downgrades readability and potentially other aspects. Since we are already using gRPC/Protobuf and Protobuf proactively maintains this jsonpb around std json, I think the proper default fallback should be jsonpb. 

I understand there's certain perf degrade in jsonpb as it's built on top of json, but I believe that should be trivial in whole request handling process. bottleneck is more at on storage level or remote dep, not a local serialiser.

Also understand there'd be other services started with std json as default so may worry about such assumption level change. Alternatively I can work to make this defaults configurable, if that works.

Reply:
as per discussion in slack, jsonpb unmarshal will break if the client passes a JSON string serialized by JSON marshaller and thus we go patch1 only first. Later incrementally rollout fallbacking to jsonpb unmarshaller logic